packages/visual-editor/src/utils/simulateMouseEvent.ts

```diff
@@ -42,3 +42,29 @@ export function simulateMouseEvent(coordX: number, coordY: number, eventName = '
   const event = new MouseEvent(eventName, options);
   element.dispatchEvent(event);
 }
+
+export function simulateComponentMove(
+  coordX: number,
+  coordY: number,
+  selector: string,
+  eventName = 'mousemove',
+) {
+  const element = document.querySelector(selector);
+
+  const options = {
+    bubbles: true,
+    cancelable: true,
+    view: window,
+    pageX: 0,
+    pageY: 0,
+    clientX: coordX - DRAGGABLE_WIDTH / 2,
+    clientY: coordY - DRAGGABLE_HEIGHT / 2 - window.scrollY,
+  };
+
+  if (!element) {
+    return;
+  }
+
+  const event = new MouseEvent(eventName, options);
+  element.dispatchEvent(event);
```

Comment:
It looks like the `simulateMouseEvent` method does most of the same logic. Should that method now call `simulateComponentMove` to avoid duplication?

```ts
export function simulateMouseEvent(coordX: number, coordY: number, eventName = 'mousemove') {
  if (!dragState.isDragStart) {
    return;
  }
  if (!dragState.isDragging) {
    updateDraggableElement(coordX, coordY);
    eventName = 'mousedown';
    dragState.updateIsDragging(true);
  }
  simulateComponentMove(coordX, coordY, '#item', eventName);
}
```